core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -161,16 +161,20 @@ private void cacheChanges() {
     List<ManifestEntry> adds = Lists.newArrayList();
     List<ManifestEntry> deletes = Lists.newArrayList();
 
-    for (ManifestEntry entry : entries(fileSchema.select(CHANGE_COLUNNS))) {
-      switch (entry.status()) {
-        case ADDED:
-          adds.add(entry.copyWithoutStats());
-          break;
-        case DELETED:
-          deletes.add(entry.copyWithoutStats());
-          break;
-        default:
+    try (CloseableIterable<ManifestEntry> entries = entries(fileSchema.select(CHANGE_COLUNNS))) {
```

Comment:
What is the purpose for `ManifestReader#cacheChanges()`. It seems previously it was used in `Basesnapshot#cacheChanges()` but now is it even used anywhere? .

Reply:
Also `ManifestReader#[addedFiles(), deletedFiles()]`

Reply:
The intent is the same as for Snapshot: to make the added and deleted files easily accessible. Manifests are `Iterable<DataFile>`, and don't expose the internal `ManifestEntry` data. That means we need to make the added and deleted files accessible through the API somehow.